tests/CollectionTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Stripe;
+
+class CollectionTest extends TestCase
+{
+    public function testAutoPagingOnePage()
+    {
+        $collection = Collection::constructFrom(array(
+            'object' => 'list',
+            'url' => '/my/path',
+            'data' => array(
+                array('id' => 'foo')
+            )
+        ), new Util\RequestOptions());
+
+        $seen = array();
+        foreach ($collection->autoPagingIterator() as $item) {
+            array_push($seen, $item['id']);
+        }
+
+        $this->assertSame($seen, array('foo'));
+    }
+
+    public function testAutoPagingTwoPages()
+    {
+        $collection = Collection::constructFrom(array(
+            'object' => 'list',
+            'url' => '/my/path',
+            'data' => array(
+                array('id' => 'foo')
+            ),
+            'has_more' => true
+        ), new Util\RequestOptions());
```

Comment:
Hey @olivierbellone! Great fixes. This is looking really good to me.

Sorry to push it, but can I make one more requested it change? It might be advantageous if we could send in an additional request option in the initial collection initialization here, and just verify that it appears in the second request below. This has the advantage of checking that parameters that were given to the original collection will be sent along in subsequent requests as expected.

I remember that I messed up something along these lines in the Java bindings and any additional sanity checks that we can put in would probably be a reasonably good idea.